doc/ota_updates/ota_updates.md

```diff
@@ -18,13 +18,44 @@ OTA may be done from:
 
-The following chapters provide more details and both methods of doing OTA.
+There is no imposed security on OTA process from being hacked. It is up to developer to ensure that updates are allowed only from legitimate / trusted source. Once update is complete module restarts and new code is executed. Developer should ensure that application running on module is shut down and restarted in safe manner. Chapters below provide additinal information  regarding security and safety of OTA process.
+
+### Security
+
+Module has to be exposed wirelessly to get it updated with a new code. That poses chances of module being violently hacked and loaded with some other firmware. To reduce likelihood of being hacked consider protecting your uploads with a password, selecting certain OTA port, etc. 
+
+Check functionality provided with [ArduinoOTA](https://github.com/esp8266/Arduino/tree/master/libraries/ArduinoOTA) library that may improve security:
+```cpp
+void setPort(uint16_t port);
+void setHostname(const char *hostname);
+void setPassword(const char *password);
+```
+If possible implement other means of protection from being hacked, e.g. exposing module for uploads only according to specific schedule, trigger OTA only be user pressing dedicated “Update” button, etc.
+
+### Safety
+
+OTA process takes ESP’s resources and bandwidth during upload. Then module is restarted and a new sketch executed. Analyse and test how it affects functionality of your existing and new sketch. 
+
+If ESP is placed in remote location and controlling some equipment, you should put additional attention what happens if operation of this equipment is suddenly interrupted by update process. Therefore decide how to put this equipment into safe state before starting the update.  For instance your module may be controlling a garden watering system in a sequence. If this sequence is not properly shut down and a water valve left open, your garden may be flooded if this valve is not closed after OTA is finished and module restarts.  
+
+The following functions are provided with [ArduinoOTA](https://github.com/esp8266/Arduino/tree/master/libraries/ArduinoOTA) library and intended to handle functionality of your application during specific stages of OTA on or on an OTA error:
+```cpp
+void onStart(OTA_CALLBACK(fn));
+void onEnd(OTA_CALLBACK(fn));
+void onProgress(OTA_CALLBACK_PROGRESS(fn));
+void onError(OTA_CALLBACK_ERROR (fn));
+```
+
+The following chapters provide more details and specific methods of doing OTA.
 
 
 ## Basic Requirements
 
 - Flash chip size is 2x the size of the sketch.
 
+Two separate, same size memory locations are used alternately. While existing sketch is executed from one location, the new sketch is saved to the other. After reset the new sketch is executed directly from where saved. Location of old sketch will be used for next OTA upload.
```

Comment:
the sketch is copied over the old one at startup.

take a look here:
https://github.com/esp8266/Arduino/blob/master/cores/esp8266/Updater.cpp#L151-L156
https://github.com/esp8266/Arduino/tree/master/bootloaders/eboot